src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py

```diff
@@ -874,6 +894,11 @@ def set_vmss_extension(
 
     # pylint: disable=no-member
     version = _normalize_extension_version(publisher, extension_name, version, vmss.location)
+    extension_profile = vmss.virtual_machine_profile.extension_profile
+    if extension_profile:
+        extensions = extension_profile.extensions
+        if extensions:
+            extension_profile.extensions = [x for x in extensions if x.type.lower() != extension_name.lower() or x.publisher.lower() != publisher.lower()]  # pylint: disable=line-too-long
```

Comment:
If I have an extension on a scaleset of 5 VMs, does that mean the VMSS has five extensions?  Is that why this is different from the VM scenario?